### PR TITLE
Tooling

### PR DESCRIPTION
**Pull request by steven-murray**
This adds a number of new tooling/package restructure things:

1. Puts the package in `src/`
2. Uses `setup.cfg` instead of `setup.py` primarily
3. Adds a bunch of `flake8` checks and plugins to both pre-commit and tox
4. Adds a skeleton `tests/` directory and TRAVIS tooling
5. Makes changes to files consistent with the flake8 checks
6. Adds a `docs/` directory and skeleton, but no actual docs
7. Adds AUTHORS and CHANGELOG, but no CONTRIBUTING
8. Adds some instructions to the README for development (probably should be moved to CONTRIBUTING when that is done).
9. Sets up tox
10. Sets up coveragerc

The main thing to be checked here are the `flake8` plugins and options -- whether we want all these (or more!). 

Addresses #21 
Addresses #11

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+=========
+Changelog
+=========
+
+Version 0.2dev
```
Comment (plaplant) [inline CHANGELOG.rst]: Given the release we just made, we should probably separate this out into a `Version 0.1.1`, which is easily installable, and then have an `Unreleased version` that turns into `0.2.0` when we've got tests going.

File: README.rst [hunk 2]
```diff
@@ -53,4 +59,42 @@ Developer Install
 If you are developing ``RIMEz``, clone the repo, install the dependencies as
 above, and then do (in the repo)::
 
-  $ pip install -e .[dev]
+  $ pip install -e ".[all,dev]"
+
+
+Optional Extras
+---------------
+There are a number of optional extras that can be installed along with `RIMEz`,
```
Comment (plaplant) [inline README.rst]: Because this is a `.rst` file, you need double-backticks to get inline code formatting.

**Comment by plaplant** [comment]
@steven-murray looks like there are some merge conflicts that need to be resolved as well.

File: CHANGELOG.rst (hunk 1 above)
Comment (steven-murray) [reply CHANGELOG.rst]: Agreed. Fixed.

File: README.rst (hunk 2 above)
Comment (steven-murray) [reply README.rst]: Ahhh yeah..

**Pull request merged**